futures-sink/src/lib.rs

```diff
@@ -114,101 +114,70 @@ pub trait Sink {
+    /// call to `start_send` to ensure that the Sink is ready to receive values.
     ///
-    /// If the sink encounters an error other than being temporarily full, it
-    /// uses the `Err` variant to signal that error. In most cases, such errors
-    /// mean that the sink will permanently be unable to receive items.
-    ///
-    /// # Panics
-    ///
-    /// This method may panic in a few situations, depending on the specific
-    /// sink:
-    ///
-    /// - It is called outside of the context of a task.
-    /// - A previous call to `start_send` or `flush` yielded an error.
+    /// In most cases, if the sink encounters an error, the sink will
+    /// permanently be unable to receive items.
     fn start_send(&mut self, item: Self::SinkItem)
                   -> Result<(), Self::SinkError>;
 
     /// Flush all output from this sink, if necessary.
     ///
-    /// Some sinks may buffer intermediate data as an optimization to improve
-    /// throughput. In other words, if a sink has a corresponding receiver then
-    /// a successful `start_send` above may not guarantee that the value is
-    /// actually ready to be received by the receiver. This function is intended
-    /// to be used to ensure that values do indeed make their way to the
-    /// receiver.
-    ///
-    /// This function will attempt to process any pending requests on behalf of
-    /// the sink and drive it to completion.
-    ///
-    /// # Return value
-    ///
     /// Returns `Ok(Async::Ready(()))` when no buffered items remain. If this
     /// value is returned then it is guaranteed that all previous values sent
-    /// via `start_send` will be guaranteed to be available to a listening
-    /// receiver.
+    /// via `start_send` have been flushed.
     ///
     /// Returns `Ok(Async::Pending)` if there is more work left to do, in which
-    /// case the current task is scheduled to wake up when more progress may be
-    /// possible.
-    ///
-    /// # Errors
-    ///
-    /// Returns `Err` if the sink encounters an error while processing one of
-    /// its pending requests. Due to the buffered nature of requests, it is not
-    /// generally possible to correlate the error with a particular request. As
-    /// with `start_send`, these errors are generally "fatal" for continued use
-    /// of the sink.
-    ///
-    /// # Panics
-    ///
-    /// This method may panic in a few situations, depending on the specific sink:
+    /// case the current task is scheduled (via `cx.waker()`) to wake up when
+    /// `poll_flush` should be called again.
     ///
-    /// - It is called outside of the context of a task.
-    /// - A previous call to `start_send` or `flush` yielded an error.
+    /// In most cases, if the sink encounters an error, the sink will
+    /// permanently be unable to receive items.
+    fn poll_flush(&mut self, cx: &mut task::Context) -> Poll<(), Self::SinkError>;
+
+    /// Flush any remaining output and close this sink, if necessary.
```

Comment:
the "if necessary" here is a bit confusing

Reply:
Done.